orchestrator/packages/api/src/controllers/docker-image-controller.ts

```diff
@@ -8,10 +8,17 @@ import {
 import { GradingQueueOperationException, enqueueImageBuild, imageIsAwaitingBuild, imageIsBeingBuilt } from "@codegrade-orca/db";
 
 export const createGraderImage = async (req: Request, res: Response) => {
-  if (!validations.graderImageBuildRequest(req.body)) {
-    return errorResponse(res, 400, [
+  const validator = validations.graderImageBuildRequest;
+  if (!validator(req.body)) {
+    const errors = [
       "The request body to build a grader image is invalid.",
-    ]);
+      ...validator.errors ?
```

Comment:
Operator precedence here is _very_ weird.  I am not sure whether the spread operator is higher or lower precedence than the ternary.  Would likely be clearer written as `...((validator.errors ?? []).map(...).filter(...))

The `!!length` is a bit obscure here, since you're working with strings, so probably clearer to write `s !== '')`

And then since you've already triplicated this code, this should be a helper function in utils.ts alongside `formatValidationError`, and the whole thing should collapse down to `return errorResponse(res, 400, formatValidationErrors("The request body to build a grader image is invalid.", validator.errors))`

```
const formatValidationErrors(msg : string, errors : undefined | unknown[]) {
  return [
    msg,
    ...((errors ?? []).map(...).filter(s => s !== '')
  ];
}
```